core/src/main/java/org/commonjava/indy/core/ctl/NfcController.java

```diff
@@ -92,14 +122,23 @@ public NotFoundCacheDTO getMissing( final StoreKey key )
             }
             catch ( final IndyDataException e )
             {
-                throw new IndyWorkflowException( "Failed to retrieve concrete constituent ArtifactStores for: %s.", e,
-                                                  key );
+                throw new IndyWorkflowException( "Failed to retrieve concrete constituent for: %s.", e, key );
             }
 
+            //Warn: This is very expensive if group holds thousands of repositories
```

Comment:
Maybe we should throw an IndyWorkflowException with an appropriate http status code if the storekey references a Group? This has potential to affect other users, if someone requests a large group.

Reply:
Ok. I have struggled with it for some time. You help me make up my mind. 👍 
After Google, I use code 422, IndyWorkflowException( SC_UNPROCESSABLE_ENTITY, ...)